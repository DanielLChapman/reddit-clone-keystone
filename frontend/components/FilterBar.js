import React, { Component } from 'react';
import { FaHotjar, FaBahai, FaGopuram, FaChartLine} from 'react-icons/fa';
import Dropdown from './Dropdown';



class FilterBar extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        let style = {
            display: "none"
        };
        if (this.props.filterState === 'Top') {
            /*style= {
                display: "inline-block"
            };*/
        }
        let options = ['Best', 'New', 'Top', 'Hot'];
        return (
            <>
                <section className="filter-bar"> 
                    <h5>{this.props.filterState} Posts</h5>
                    <ul>
                        <li data-testid="best-filter-bar" aria-disabled={this.props.filterState === 'Best'} className={this.props.filterState === 'Best' ? 'active' : null} onClick={() => this.props.setFilterState('Best')}><FaHotjar /> Best</li>
                        <li aria-disabled={this.props.filterState === 'New'} className={this.props.filterState === 'New' ? 'active' : null} onClick={() => this.props.setFilterState('New')}><FaBahai /> New</li>
                        <li data-testid="top-filter-bar" aria-disabled={this.props.filterState === 'Top'} className={this.props.filterState === 'Top' ? 'active' : null} onClick={() => this.props.setFilterState('Top')}><FaGopuram /> Top</li>
                        <li className="time-selector-for-filter-bar" style={style}>Today</li>
                        <li aria-disabled={this.props.filterState === 'Hot'} className={this.props.filterState === 'Hot' ? 'active' : null} onClick={() => this.props.setFilterState('Hot')}><FaChartLine /> Hot</li>
                    </ul>
                </section>
                <div className="filter-dropdown">
                    <Dropdown returnFunc={this.props.setFilterState} selected={this.props.filterState} options={options} />
                </div>
            </>
        );
    }
}

export default FilterBar;