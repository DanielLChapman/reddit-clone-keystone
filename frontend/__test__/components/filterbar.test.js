import { render, screen } from '@testing-library/react';
import {MockedProvider} from '@apollo/react-testing';
import FilterBar from '../../components/FilterBar';


describe('It loads the filter bar', () => {
    it('loads filter state with Best and expects best to have an active class', () => {
        const { container, debug } = render(
            <MockedProvider >
              <FilterBar filterState="Best" setFilterState={() => {console.log('here')}}/>
            </MockedProvider>
          );

          //expect best to have the active class and an aria-disabled

          let identifier = screen.getByTestId('best-filter-bar');
          expect(identifier).toHaveAttribute('aria-disabled')
          expect(container).toMatchSnapshot();



    });

    it('loads a different filter state and expects changes', () => {
        const { container, debug } = render(
            <MockedProvider >
              <FilterBar filterState="Top" setFilterState={() => {console.log('here')}}/>
            </MockedProvider>
          );

          let identifier = screen.getByTestId('best-filter-bar');
          expect(identifier).not.toBeDisabled();

          let identifier2 = screen.getByTestId('top-filter-bar');
          expect(identifier2).toHaveAttribute('aria-disabled');
    })
})