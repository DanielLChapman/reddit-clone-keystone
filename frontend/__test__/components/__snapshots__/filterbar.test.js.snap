// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It loads the filter bar loads filter state with Best and expects best to have an active class 1`] = `
<div>
  <section
    class="filter-bar"
  >
    <h5>
      Best
       Posts
    </h5>
    <ul>
      <li
        aria-disabled="true"
        class="active"
        data-testid="best-filter-bar"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M414.9 161.5C340.2 29 121.1 0 121.1 0S222.2 110.4 93 197.7C11.3 252.8-21 324.4 14 402.6c26.8 59.9 83.5 84.3 144.6 93.4-29.2-55.1-6.6-122.4-4.1-129.6 57.1 86.4 165 0 110.8-93.9 71 15.4 81.6 138.6 27.1 215.5 80.5-25.3 134.1-88.9 148.8-145.6 15.5-59.3 3.7-127.9-26.3-180.9z"
          />
        </svg>
         Best
      </li>
      <li
        aria-disabled="false"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M496.25 202.52l-110-15.44 41.82-104.34c6.67-16.64-11.6-32.18-26.59-22.63L307.44 120 273.35 12.82C270.64 4.27 263.32 0 256 0c-7.32 0-14.64 4.27-17.35 12.82l-34.09 107.19-94.04-59.89c-14.99-9.55-33.25 5.99-26.59 22.63l41.82 104.34-110 15.43c-17.54 2.46-21.68 26.27-6.03 34.67l98.16 52.66-74.48 83.54c-10.92 12.25-1.72 30.93 13.29 30.93 1.31 0 2.67-.14 4.07-.45l108.57-23.65-4.11 112.55c-.43 11.65 8.87 19.22 18.41 19.22 5.15 0 10.39-2.21 14.2-7.18l68.18-88.9 68.18 88.9c3.81 4.97 9.04 7.18 14.2 7.18 9.54 0 18.84-7.57 18.41-19.22l-4.11-112.55 108.57 23.65c17.36 3.76 29.21-17.2 17.35-30.49l-74.48-83.54 98.16-52.66c15.64-8.39 11.5-32.2-6.04-34.66zM338.51 311.68l-51.89-11.3 1.97 53.79L256 311.68l-32.59 42.49 1.96-53.79-51.89 11.3 35.6-39.93-46.92-25.17 52.57-7.38-19.99-49.87 44.95 28.62L256 166.72l16.29 51.23 44.95-28.62-19.99 49.87 52.57 7.38-46.92 25.17 35.61 39.93z"
          />
        </svg>
         New
      </li>
      <li
        aria-disabled="false"
        data-testid="top-filter-bar"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M496 352h-16V240c0-8.8-7.2-16-16-16h-16v-80c0-8.8-7.2-16-16-16h-16V16c0-8.8-7.2-16-16-16s-16 7.2-16 16v16h-64V16c0-8.8-7.2-16-16-16s-16 7.2-16 16v16h-64V16c0-8.8-7.2-16-16-16s-16 7.2-16 16v16h-64V16c0-8.8-7.2-16-16-16S96 7.2 96 16v112H80c-8.8 0-16 7.2-16 16v80H48c-8.8 0-16 7.2-16 16v112H16c-8.8 0-16 7.2-16 16v128c0 8.8 7.2 16 16 16h80V352h32V224h32v-96h32v96h-32v128h-32v160h80v-80c0-8.8 7.2-16 16-16h64c8.8 0 16 7.2 16 16v80h80V352h-32V224h-32v-96h32v96h32v128h32v160h80c8.8 0 16-7.2 16-16V368c0-8.8-7.2-16-16-16zM232 176c0-8.8 7.2-16 16-16h16c8.8 0 16 7.2 16 16v48h-48zm56 176h-64v-64c0-8.8 7.2-16 16-16h32c8.8 0 16 7.2 16 16z"
          />
        </svg>
         Top
      </li>
      <li
        class="time-selector-for-filter-bar"
        style="display: none;"
      >
        Today
      </li>
      <li
        aria-disabled="false"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M496 384H64V80c0-8.84-7.16-16-16-16H16C7.16 64 0 71.16 0 80v336c0 17.67 14.33 32 32 32h464c8.84 0 16-7.16 16-16v-32c0-8.84-7.16-16-16-16zM464 96H345.94c-21.38 0-32.09 25.85-16.97 40.97l32.4 32.4L288 242.75l-73.37-73.37c-12.5-12.5-32.76-12.5-45.25 0l-68.69 68.69c-6.25 6.25-6.25 16.38 0 22.63l22.62 22.62c6.25 6.25 16.38 6.25 22.63 0L192 237.25l73.37 73.37c12.5 12.5 32.76 12.5 45.25 0l96-96 32.4 32.4c15.12 15.12 40.97 4.41 40.97-16.97V112c.01-8.84-7.15-16-15.99-16z"
          />
        </svg>
         Hot
      </li>
    </ul>
  </section>
  <div
    class="filter-dropdown"
  >
    <select>
      <option
        value="Best"
      >
        Best
      </option>
      <option
        value="New"
      >
        New
      </option>
      <option
        value="Top"
      >
        Top
      </option>
      <option
        value="Hot"
      >
        Hot
      </option>
    </select>
  </div>
</div>
`;
